File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.ts
function subarraySum(nums: number[], k: number): number {
    const map = new Map();
    map.set(0, 1);
    let ret = 0;
    let curr = 0;
    
    for(let i of nums){
        curr += i;
        ret += map.get(curr-k) || 0;
        map.set(curr, (map.get(curr) || 0) + 1);        
    }
    return ret;
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cs
public class Solution {
    public int SubarraySum(int[] nums, int k) {
        int ret = 0, curr = 0;
        Dictionary<int, int> dic = new Dictionary<int, int>{[0] = 1};
        
        foreach(int n in nums){
            curr += n; 
            ret += dic.GetValueOrDefault(curr-k, 0);
            dic[curr] = dic.GetValueOrDefault(curr, 0) + 1;
        }        
        return ret;
    }
}